ros-rviz-image.html

```diff
@@ -116,10 +117,34 @@
           if (updatingImage) {
             return;   // Drop frame if previous one didn't load yet.
           }
+
           updatingImage = true;
           let image = new Image();
-          image.src = "data:image/jpg;base64," + message.data;
+
+          let blobUrl = null;
+          if (message.data.buffer) {
+            // Binary path -- load Blob
+            blobUrl = URL.createObjectURL(new Blob([message.data], {type: 'image/jpg'}));
+            image.src = blobUrl;
+          } else {
+            // Text path -- load base64 string
+            image.src = "data:image/jpg;base64," + message.data;
+          }
+
+          function finishUpdate() {
+            updatingImage = false;
+            if (blobUrl) {
+              URL.revokeObjectURL(blobUrl);
+            }
+          }
+
+          image.onerror = function(e) {
+            finishUpdate();
+            console.error("Error loading Image", e);
+          };
+
           image.onload = function() {
+            finishUpdate();
```

Comment:
I don't know if this makes a huge difference, but maybe this call should be at the end after drawing the image so as to block the flag until everything is done (at least for the text path it would make sense IMO).

Reply:
I put it at the beginning to ensure the state gets reset before an exception could potentially prevent it from happening.  Otherwise there is no chance of a race, JavaScript will not preempt any of these handlers.